src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -93,22 +154,91 @@ public function isRisky()
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        $functionNames = $this->getFunctionNames();
+        $functionFilter = $this->getFunctionFilter();
+
+        if ('global' === $this->configuration['scope']) {
+            $this->fixFunctionCalls($tokens, $functionFilter, 0, \count($tokens) - 1);
+
+            return;
+        }
+
+        // 'scope' is 'namespaced' here
+        foreach (\array_reverse($this->getUserDefinedNamespaces($tokens)) as $namespace) {
+            $this->fixFunctionCalls($tokens, $functionFilter, $namespace['open'], $namespace['close']);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('exclude', 'List of functions to ignore.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([static function (array $value) {
+                    foreach ($value as $functionName) {
+                        if (!\is_string($functionName) || '' === \trim($functionName) || \trim($functionName) !== $functionName) {
+                            throw new InvalidOptionsException(\sprintf(
+                                'Each element must be a non-empty, trimmed string, got "%s" instead.',
+                                \is_object($functionName) ? \get_class($functionName) : \gettype($functionName)
+                            ));
+                        }
+                    }
+
+                    return true;
+                }])
+                ->setDefault([])
+                ->getOption(),
+            (new FixerOptionBuilder('include', 'List of function names or sets to fix. Defined sets are `@internal` (all native functions), `@all` (all global functions) and `@compiler_optimized` (functions that are specially optimized by Zend).'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([static function (array $value) {
+                    foreach ($value as $functionName) {
+                        if (!\is_string($functionName) || '' === \trim($functionName) || \trim($functionName) !== $functionName) {
+                            throw new InvalidOptionsException(\sprintf(
+                                'Each element must be a non-empty, trimmed string, got "%s" instead.',
+                                \is_object($functionName) ? \get_class($functionName) : \gettype($functionName)
+                            ));
+                        }
 
-        $indexes = [];
+                        static $sets = [
```

Comment:
no need for static, php7 will optimize it anyway, and we don't need to store it in memory indefinitely

Reply:
thanks, fixed :+1: